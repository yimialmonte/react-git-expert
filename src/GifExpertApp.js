import React, { useState } from 'react';
import { AddCategory } from './components/AddCategoy';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
