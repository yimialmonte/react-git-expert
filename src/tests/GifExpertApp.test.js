import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('Test for component GifExpertApp', () => {

  test('Show show component correctly', () => {
    const wraper = shallow(<GifExpertApp />);
    expect(wraper).toMatchSnapshot();
  });

  test('Should show a list of items', () => {
    const categories = ['Michael', 'Lebron']
    const wraper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wraper).toMatchSnapshot();
    expect(wraper.find('GifGrid').length).toBe(categories.length);
  })
});
