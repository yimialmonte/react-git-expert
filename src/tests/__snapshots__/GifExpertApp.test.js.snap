// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component GifExpertApp Should show a list of items 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Michael"
      key="Michael"
    />
    <GifGrid
      category="Lebron"
      key="Lebron"
    />
  </ol>
</Fragment>
`;

exports[`Test for component GifExpertApp Show show component correctly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
